src/adaXT/decision_tree/DepthTreeBuilder.py

```diff
@@ -88,6 +88,10 @@ def __init__(
         else:
             self.splitter = Splitter(self.features, self.response, criteria)
 
+        _, col = X.shape
+        self.feature_indices = np.arange(col, dtype=np.int32)
+        self.num_features = col
```

Comment:
Should probably be changed to num_tot_features, as the max_features could be less than this value.

Reply:
Done.